pallets/author-inherent/src/lib.rs

```diff
@@ -125,6 +126,17 @@ decl_module! {
 	}
 }
 
+impl<T: Config> FindAuthor<T::AccountId> for Module<T> {
+	fn find_author<'a, I>(_digests: I) -> Option<T::AccountId>
+	where
+		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+	{
+		// We don't use the digests at all.
+		// This will only return the correct author _after_ the authorship inherent is processed.
```

Comment:
Could there be a risk of the pallet ethereum trying to use it before ?

Reply:
I don't think so, pallet-ethereum uses it `on_finalize`.

Reply:
No. Pallet ethereum generates its ethereum-style block at the end of block execution. Inherents are the first extrinsics during the block execution.

The only thing that could trip this case is if another extrinsic that is includued earlier than the author inherent needs it. And since the nodes insert inherents first, it could only be other inherents that need it. Neither the timestamp nor validation inherents need this information, so we are safe :)